src/agent/common.ts

```diff
@@ -453,24 +452,23 @@ function createMaskFunction(jobEnvironment: agentifm.JobEnvironment): Replacemen
                     return results;
                 });
             }
-        });
-
-        endpoints.forEach((endpoint: agentifm.ServiceEndpoint) => {
-            if (endpoint.authorization != null && endpoint.authorization['parameters'] != null) {
-                var keys = Object.keys(endpoint.authorization['parameters']);
-                keys.forEach((key: string) => {
-                    var valueToReplace: string = endpoint.authorization['parameters'][key];
-                    if (valueToReplace === null || valueToReplace === undefined) {
-                        indexFunctions.push((input: string) => {
-                            var results: ReplacementPosition[] = [];
-                            var index: number = input.indexOf(valueToReplace);
-                            while (index > -1) {
-                                results.push({ start: index, length: valueToReplace.length });
-                                index = input.indexOf(valueToReplace, index + 1);
-                            }
-                            return results;
-                        });
+            else if (maskHint.type === agentifm.MaskType.Regex) {
+                indexFunctions.push((input: string) => {
+                    var results: ReplacementPosition[] = [];
+                    var actualIndex = 0;
+                    while (input.length > 0) {
+                        var match = input.match(maskHint.value);
```

Comment:
There's a check somewhere else to make sure maskHint.value is not an empty string, right?